homestar-runtime/src/cli/init.rs

```diff
@@ -0,0 +1,317 @@
+        }
+    }
+}
+
+/// Handle the `init` command.
+pub fn handle_init_command(
+    output_mode: OutputMode,
+    key_arg: Option<KeyArg>,
+    key_type_arg: Option<KeyTypeArg>,
+    quiet: bool,
+    no_input: bool,
+) -> Result<()> {
+    let mut settings_builder = SettingsBuilder::default();
+    let mut node_builder = NodeBuilder::default();
+    let mut network_builder = NetworkBuilder::default();
+
+    let mut writer = handle_quiet(quiet)?;
+    let key_type = handle_key_type(key_type_arg, no_input, &mut writer)?;
+    let keypair_config = handle_key(key_arg, key_type, no_input, &mut writer)?;
+
+    let network = network_builder
+        .keypair_config(keypair_config)
+        .build()
+        .expect("to build network");
+
+    let node = node_builder
+        .network(network)
+        .build()
+        .expect("to build node");
+
+    let settings = settings_builder
+        .node(node)
+        .build()
+        .expect("to builder settings");
+
+    let settings_toml = toml::to_string_pretty(&settings).expect("to serialize settings");
+
+    handle_output_mode(output_mode, no_input, &mut writer)?
+        .write_all(settings_toml.as_bytes())
+        .expect("to write settings file");
+
+    Ok(())
+}
+
+fn handle_quiet(quiet: bool) -> Result<Box<dyn Write>> {
+    if quiet {
+        Ok(Box::new(empty()))
+    } else {
+        Ok(Box::new(stdout()))
+    }
+}
+
+fn handle_output_mode(
+    output_mode: OutputMode,
+    no_input: bool,
+    writer: &mut Box<dyn Write>,
+) -> Result<Box<dyn Write>> {
+    match output_mode {
+        OutputMode::StdOut => Ok(Box::new(stdout())),
+        OutputMode::File { path, force: true } => {
+            if let Some(parent) = path.parent() {
+                std::fs::create_dir_all(parent).expect("to create parent directory");
+            }
+
+            let settings_file = File::options()
+                .read(true)
+                .write(true)
+                .create(true)
+                .truncate(true)
+                .open(&path)
+                .expect("to open settings file");
+
+            writeln!(writer, "Writing settings to {:?}", path).expect("to write");
+
+            Ok(Box::new(settings_file))
+        }
+        OutputMode::File { path, force: false } => {
+            if let Some(parent) = path.parent() {
+                std::fs::create_dir_all(parent).expect("to create parent directory");
+            }
+
+            let settings_file = File::options()
+                .read(true)
+                .write(true)
+                .create_new(true)
+                .open(&path);
+
+            // This seemingly convoluted match is to avoid the risk of a
+            // TOCTOU race condition, where another process creates the file
+            // in between this one checking for its existence and opening it.
+            //
+            // TODO: there should probably be a flag for non-interactive use
```

Comment:
we do have a --no-input flag for non-interactive use?

Reply:
Good catch, thanks!